wallet-new/src/Cardano/Wallet/Kernel/DB/Util/IxSet.hs

```diff
@@ -155,6 +157,16 @@ size = IxSet.size . unwrapIxSet
 getOne :: HasPrimKey a => IxSet a -> Maybe a
 getOne = fmap coerce . IxSet.getOne . unwrapIxSet
 
+-- | Project out the underlying set
+--
+-- Use with caution! This loses all the indices, potentially losing all the
+-- benefits that 'IxSet' provides.
+toMap :: HasPrimKey a => IxSet a -> Map (PrimKey a) a
+toMap = Map.mapKeysMonotonic (primKey . unwrapOrdByPrimKey)
```

Comment:
Isn't this falling into the same anti-pattern we wanted to avoid with `toList`? In my current branch I am carefully trying to avoid to introduce an `Ixset.toList` function, but this one, I guess, could be used in an evil manner to write a `toList` implementation, couldn't it?

If we think this is unavoidable, I would be inclined to take the plunge and _also_ export a `toList` function with the same caveats, as it would make my code less horrid in some places. Happy to bounce off ideas here.

Reply:
In this particular case this is precisely the format I needed, and the code is somewhat non-trivial, so I figured it belonged in the main `IxSet` wrapper. We can reconsider `toList` if you really think it's needed?